File: 백준/Silver/10815. 숫자 카드/숫자 카드.py
import sys
input = sys.stdin.readline
write = sys.stdout.write

N = input()
cards = set(map(int, input().split(" ")))
M = input()
check_cards = list(map(int, input().split(" ")))

result = ['1' if card in cards else '0' for card in check_cards]
write(' '.join(result) + '\n')

File: 백준/Silver/10815. 숫자 카드/숫자 카드.kt

fun main() {
    val br = System.`in`.bufferedReader()
    val bw = System.`out`.bufferedWriter()
    val sb = StringBuilder()
    
    val N = br.readLine()
    val cards = HashSet(br.readLine().split(" "))
    val M = br.readLine()
    val checkCards = br.readLine().split(" ")

    for(card in checkCards){
        if(card in cards){
            sb.append("1 ")
        }else{
            sb.append("0 ")
        }
    }
    bw.write(sb.toString())
    bw.flush()
    bw.close()
    br.close()
}
